extras/linked-token/src/LinkedToken.sol

```diff
@@ -58,19 +60,39 @@ abstract contract LinkedToken is IpcExchange {
 
     event LinkedTokenReceived(address indexed recipient, uint256 amount);
 
+    /// @custom:oz-upgrades-unsafe-allow constructor
+    constructor() {
+        _disableInitializers();
+    }
+
+
     /**
-     * @dev Constructor for IpcTokenController
+     * @dev Initialize function for IpcTokenController
      * @param gateway Address of the gateway for cross-network communication
      * @param underlyingToken Address of the destination contract for minting
      * @param linkedSubnet SubnetID of the destination network
      */
-    constructor(
+    function __LinkedToken_init(
         address gateway,
         address underlyingToken,
-        SubnetID memory linkedSubnet
-    ) IpcExchange(gateway) {
+        SubnetID memory linkedSubnet,
+        address linkedContract
+    ) public onlyInitializing {
+        __IpcExchangeUpgradeable_init(gateway);
         _underlying = IERC20(underlyingToken);
         _linkedSubnet = linkedSubnet;
+        _linkedContract = linkedContract;
+
+        emit LinkedTokenInitialized({
+            gateway: gateway,
+            underlying: address(_underlying),
+            linkedSubnet: _linkedSubnet,
+            linkedContract: _linkedContract
+        });
+    }
+
+    function setLinkedContract(address linkedContract) external onlyOwner {
+        _linkedContract = linkedContract;
     }
```

Comment:
Do we actually need this function now that we have upgradable contracts? Couldn't we pass the proxy address of the counterparty to the initializer, instead of passing in a zero address like we do everywhere now?

Reply:
There may be a way to do it, but the documentation that I've based our implementation off of involves deploying the proxy in the same transaction that we call the initializer so we still have the chicken and the egg problem

Reply:
Got it, thanks!